RestfulDropApi.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace DropDownloadCore
+{
+    //Http calls to drop api and azure storage uses the following apis.
+    //https://www.1eswiki.com/wiki/VSTS_Drop#Drop_REST_API
+    //https://opsstaging.www.visualstudio.com/en-us/docs/integrate/api/drop/manifest?branch=teams%2Fartifact%2Fversion2
+    //https://opsstaging.www.visualstudio.com/en-us/docs/integrate/api/blobstore/blobs?branch=teams%2Fartifact%2Fversion2
+
+    public sealed class RestfulDropApi : IDropApi
+    {
+        public static readonly string APIVersionParam = "api-version";
+        private readonly string _pat;
+        private readonly ILogger _logger;
+
+        public RestfulDropApi(string pat, ILoggerFactory loggerFactory)
+        {
+            if (string.IsNullOrWhiteSpace(pat))
+            {
+                throw new ArgumentException("Non-empty PAT required.", nameof(pat));
+            }
+
+            _pat = pat;
+
+            _logger = loggerFactory.CreateLogger<RestfulDropApi>();
+        }
+        
+        
+        /// <summary>
+        /// Gets the manifest details
+        /// </summary>
+        /// <param name="manifestUri">The manifest uri.</param>
+        /// <param name="blobAPIVersion">The API version to use for the blob API.</param>
+        /// <param name="relativeRoot">The root path relative to the drop to retrieve.</param>
+        /// <returns>The manifest details.</returns>         
+        public async Task<IEnumerable<VstsFile>> GetVstsManifest(Uri manifestUri, string blobAPIVersion, 
+                                                                 string relativeRoot)
+        {
+            // dotnet core doesn't currently handle vsts redirects well. Poking both teams about it
+            // for now disable redirects.
+            var clientHandler = new HttpClientHandler() { AllowAutoRedirect = false };
```

Comment:
apparently dotnet/aspnet has a httpclient factory which I want to poke at in the future. where you can say get me a httpclient for foo and elsewhere configure auth and base address. Hanselman mentioned it in the last week.

Reply:
can also put polly retries on that httpclient.

Reply:
That sounds pretty nice actually

Reply:
TODO in future